bittide/src/Bittide/Node.hs

```diff
@@ -319,3 +320,17 @@ unconcatC ::
 unconcatC SNat = Circuit go
  where
   go (fwd, bwd) = (concat bwd, unconcat SNat fwd)
+
+-- | Map a circuit over a vector of circuits
+vecMap :: forall n a b. Circuit a b -> Circuit (Vec n a) (Vec n b)
+vecMap (Circuit f) = Circuit go
+ where
+  go (fwds, bwds) = unzip $ zipWith (curry f) fwds bwds
+
+-- | Map a function over a Circuit of CSignals
+cSigMap ::
+  forall dom a b.
+  (KnownDomain dom) =>
+  (a -> b) ->
+  Circuit (CSignal dom a) (CSignal dom b)
+cSigMap fn = Circuit $ \(m, _) -> (pure (), fn <$> m)
```

Comment:
This function is also exists in [HwCcTopologies.hs](https://github.com/bittide/bittide-hardware/blob/8f6a4ddcb0477ec7f5f33b636b4c96d82bf64af1/bittide-instances/src/Bittide/Instances/Hitl/HwCcTopologies.hs#L219). Is it worth it to deduplicate them?

Reply:
The reason why I did not remove `DataLink` yet entirely is because that would leave the scatter unit and gather unit with not tests at the moment. I'll get that in the next PR :)

Reply:
Regarding the deduplication of `cSigMap`, we should add some form of mapping upstream to `clash-protocols`. 
Though I do not know yet if that should be a function per protocol or whether we can make a typeclass.
I would not worry right now about a function that's duplicated once